### PR TITLE
8297757: VarHandles.getStaticFieldFromBaseAndOffset should get the receiver type from VarHandle

### PR DESCRIPTION
**Pull request by mlchung**
`VarHandles.getStaticFieldFromBaseAndOffset` maps a base/offset/fieldType to a static `Field`.   It's fragile to assume that the location of a static field returned by `Unsafe.staticFieldBase` is a Class object.    This changes the VarHandle implementation for static fields (i.e. `FieldStaticReadOnly` and `FieldStaticReadWrite` classes) to include the receiver type in addition to the base and offset.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297757](https://bugs.openjdk.org/browse/JDK-8297757): VarHandles.getStaticFieldFromBaseAndOffset should get the receiver type from VarHandle

### Reviewers
 * [Paul Sandoz](https://openjdk.org/census#psandoz) (@PaulSandoz - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12100/head:pull/12100` \
`$ git checkout pull/12100`

Update a local copy of the PR: \
`$ git checkout pull/12100` \
`$ git pull https://git.openjdk.org/jdk pull/12100/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12100`

View PR using the GUI difftool: \
`$ git pr show -t 12100`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12100.diff">https://git.openjdk.org/jdk/pull/12100.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/invoke/VarHandles.java [hunk 1]
```diff
@@ -183,11 +183,10 @@ static Field getFieldFromReceiverAndOffset(Class<?> receiverType,
     }
 
     // Required by instance static field handles
-    static Field getStaticFieldFromBaseAndOffset(Object base,
+    static Field getStaticFieldFromBaseAndOffset(Class<?> receiverType,
+                                                 Object base,
```
Comment (PaulSandoz) [inline src/java.base/share/classes/java/lang/invoke/VarHandles.java:187]: `base` is now unused.

Comment (mlchung) [reply src/java.base/share/classes/java/lang/invoke/VarHandles.java]: Removed.  Thanks for catching it.

**Review by PaulSandoz — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]

**Comment by gilles-duboscq** [comment]
Thank you for this fix @mlchung!

**Pull request closed without merging**